### PR TITLE
poc: moduleFromEndpoints

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
### 🔗 Linked issue

IN-3860

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

PoC of new `middlewareModule`, a new standard way of communicating with the Server Middleware.
As it's PoC, it does not contains the docs, changesets, proper test coverage or a final file structure.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/sdk/src/modules/connect/utils/getHttpClient.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  return async (url, config) => {
+    const headers = getHeaders(config);
+    const method = config.method || "POST";
+    const fullUrl = getUrl(url, method === "GET" ? config.params : undefined);
+
+    try {
+      return await httpClient(fullUrl, {
+        ...config,
+        params: method === "POST" ? config.params : [],
```
Comment (lukasborawski) [inline packages/sdk/src/modules/connect/utils/getHttpClient.ts:82]: Why it's `[]` shouldn't this be an object?

File: packages/sdk/src/modules/connect/utils/getHttpClient.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  const defaultHttpClient: HTTPClient = async (url, config) => {
+    const response = await fetch(url, {
+      ...config,
+      body: JSON.stringify(config.params),
+      credentials: "include",
+    });
+    return response.json();
+  };
+
+  const defaultErrorHandler: ErrorHandler = (error: any) => {
```
Comment (lukasborawski) [inline packages/sdk/src/modules/connect/utils/getHttpClient.ts:68]: I believe that this particular function is unnecessary.

**Comment by WojtekTheWebDev** [comment]
It's going to be separated to smaller PRs to review it easier.

**Comment by WojtekTheWebDev** [comment]
Duplicated by: https://github.com/vuestorefront/vue-storefront/pull/7052

**Pull request closed without merging**